frame/staking/src/lib.rs

```diff
@@ -3141,7 +3202,7 @@ impl<T: Config> Pallet<T> {
 	/// Move a stash account from one bag to another, depositing an event on success.
 	///
 	/// If the stash changed bags, returns `Some((from, to))`.
-	pub fn do_rebag(stash: &T::AccountId) -> Option<(BagIdx, BagIdx)> {
+	pub fn do_rebag(stash: &T::AccountId) -> Option<(VoteWeight, VoteWeight)> {
```

Comment:
I can guess that there's no more BagIdx around, and each bag is identified by its lower weight now? 

Not a big deal, but FYI this is slightly inefficient for the rebag tx. You would have an easier time submitting two indices than two `VoteWeights`, but anyways.

Reply:
Upper weight, but yes, that's fundamentally it. Agree that it's slightly less efficient, but we had to do it like that to make it efficient to change the bags with a change of runtime.

Reply:
yeah that's well worth it IMO. re-binary-searching the bags list should not be super time consuming.